pkg/qrcodegenerator/qrcode/gateways/ttigpro1.go

```diff
@@ -0,0 +1,140 @@
+// Copyright © 2024 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gateways
+
+import (
+	"regexp"
+	"strings"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/types"
+)
+
+const (
+	formatIDttigpro1 = "ttigpro1"
+
+	euiLength        = 16
+	OwnerTokenLength = 12
+)
+
+// TTIgpro1 is a format for gateway identification QR codes.
+type TTIgpro1 struct {
+	GtwEUI     types.EUI64 // lowercase base16, 16 chars long
+	OwnerToken []byte      // base62, 12 chars long
+}
+
+// ttigpro1Regex is the regular expression to match the TTIgpro1 format.
+// The format is as follows: https://ttig.pro/c/{16 lowercase base16 chars}/{12 base62 chars}
+var ttigpro1Regex = regexp.MustCompile(`^https://ttig\.pro/c/([a-f0-9]{16})/([a-zA-Z0-9]{12})$`)
+
+// Validate implements the Data interface.
+func (m TTIgpro1) Validate() error {
+	if err := m.validateOwnerToken(m.OwnerToken); err != nil {
```

Comment:
Treat owner token as string, don't try to parse it or do anything with it other than the length check that is in the regex

Reply:
fixed